File: app/src/main/java/com/raredev/vcspace/activities/SettingsActivity.java
package com.raredev.vcspace.activities;

import static com.raredev.vcspace.res.R.string;

import android.os.Bundle;
import android.view.View;
import com.raredev.vcspace.R;
import com.raredev.vcspace.databinding.ActivitySettingsBinding;
import com.raredev.vcspace.fragments.SettingsFragment;

public class SettingsActivity extends BaseActivity {
  private ActivitySettingsBinding binding;

  @Override
  protected View getLayout() {
    binding = ActivitySettingsBinding.inflate(getLayoutInflater());
    return binding.getRoot();
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setSupportActionBar(binding.toolbar);
    getSupportActionBar().setTitle(string.menu_settings);
    binding.toolbar.setNavigationOnClickListener((v) -> onBackPressed());

    if (getSupportFragmentManager().findFragmentByTag(SettingsFragment.TAG) == null) {
      getSupportFragmentManager()
          .beginTransaction()
          .add(R.id.settings_container, new SettingsFragment(), SettingsFragment.TAG)
          .commit();
    }
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    binding = null;
  }
}

File: app/src/main/java/com/raredev/vcspace/activities/SettingsActivity.kt
package com.raredev.vcspace.activities

import android.os.Bundle
import android.view.View
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.NavigationUI
import com.google.android.material.R.attr
import com.raredev.vcspace.databinding.ActivitySettingsBinding
import com.raredev.vcspace.extensions.getAttrColor

class SettingsActivity : BaseActivity() {

  private var _binding: ActivitySettingsBinding? = null
  private val binding: ActivitySettingsBinding
    get() = checkNotNull(_binding)

  override val navigationBarDividerColor: Int
    get() = getAttrColor(attr.colorOutlineVariant)

  override fun getLayout(): View {
    _binding = ActivitySettingsBinding.inflate(layoutInflater)
    return binding.root
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setSupportActionBar(binding.toolbar)

    val navHostFragment =
      supportFragmentManager.findFragmentById(binding.navHostFragment.id) as NavHostFragment
    NavigationUI.setupWithNavController(binding.toolbar, navHostFragment.navController)
  }

  override fun onDestroy() {
    super.onDestroy()
    _binding = null
  }
}
